api/algorithms/supervised/linear_regression.py

```diff
@@ -0,0 +1,325 @@
+# algorithms/supervised/linear_regression.py
+"""
+This module contains implementation of the Linear Regression algorithm.
+"""
+import logging
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+from algorithms.base.supervised import SupervisedAlgorithm
+from schemas.configs.algorithms_configs import LinearRegressionParams
+from utils.losses import LossFunction, MeanAbsoluteError, MeanSquaredError
+from utils.metrics import mean_absolute_error, mean_squared_error
```

Comment:
This should read:
from api.algorithms.base.supervised import SupervisedAlgorithm
from api.schemas.configs.algorithms_configs import LinearRegressionParams
from api.utils.losses import LossFunction, MeanAbsoluteError, MeanSquaredError
from api.utils.metrics import mean_absolute_error, mean_squared_error